app/components/navigation/navigation-item.tsx

```diff
@@ -0,0 +1,10 @@
+import { useState } from "react";
+import INavigationItem from '@/app/models/INavigationItem';
+export default function NavigationItem({ inputAttr, labelAttr, listAttr, children, hasSubpages }: INavigationItem) {
+    const [isHidden, setIsHidden] = useState(true);
+    return (<>
+        <input id={inputAttr.id} className={inputAttr.classes} type="checkbox" onChange={(e)=> setIsHidden(!e.target.value)}/>
```

Comment:
nit: I would say formatting this so that it can be more friendly to read on narrow screens can be a nice optimization in code quality and review. It can also make diffs easier to read.
```suggestion
        <input
            id={inputAttr.id}
            className={inputAttr.classes}
            type="checkbox"
            onChange={(e)=> setIsHidden(!e.target.value)}
        />
```

Reply:
> nit: I would say formatting this so that it can be more friendly to read on narrow screens can be a nice optimization in code quality and review. It can also make diffs easier to read.

Agreed. I'll make this change here. I've also created an issue to tackle stuff like this with auto-formatting: https://github.com/jjm3x3/AmazingRaceFantasy/issues/97